File: modules/react-native-ios-tab-bar/android/src/main/java/expo/modules/iostabbar/ReactNativeIosTabBarModule.kt
package expo.modules.iostabbar

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ReactNativeIosTabBarModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ReactNativeIosTabBar")

    View(ReactNativeIosTabBarView::class) {
    }
  }
}

File: modules/react-native-ios-tab-bar/ios/ReactNativeIosTabBarModule.swift
import ExpoModulesCore

public class ReactNativeIosTabBarModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ReactNativeIosTabBar")

        View(ReactNativeIosTabBarView.self) {
            Prop("tabs") { (view: ReactNativeIosTabBarView, tabs: [[String: String]]) in
                view.setTabs(tabs)
            }

            Prop("activeIndex") { (view: ReactNativeIosTabBarView, index: Int) in
                view.setActiveIndex(index)
            }

            Prop("tintColor") { (view: ReactNativeIosTabBarView, color: String) in
                view.setTintColor(color)
            }

            Events("onTabChange")
        }
    }
}